services/identity-service/src/did/did.service.ts

```diff
@@ -8,21 +8,43 @@ import { VaultService } from '../utils/vault.service';
 import { Identity } from '@prisma/client';
 @Injectable()
 export class DidService {
+  static getKeySignType = (algo: string): any => {
+    switch (algo) {
+      case 'Ed25519':
+      case 'EdDSA':
+        return {
+          keyType: "JsonWebKey2020",
+          signType: "JsonWebSignature2020"
+        };
+      case 'secp256k1':
```

Comment:
Same with secp256k1 and ES256K. For now you can set it to JsonWebKey2020 and JsonWebSignature2020 for all. Keep this Switch Case but just return the same object for every type. 

The Actual KeyType is fetched from CRV section of the publicKeyJwk.